### PR TITLE
Adding tutorial on Qibocal API in documentation

### LINKED ISSUE #573: Add Qibocal API tutorial in doc

### PR DESCRIPTION
**Pull request by andrea-pasquale**
Closes #573.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

File: doc/source/tutorials/api.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+documentation of the specific protocol or by simply inspecting ``protocol.parameters_type``.
+For ``single_shot_classification`` we can pass just the number of shots
+in the following way:
+
+.. code-block:: python
+
+    parameters = experiment.parameters_type.load(dict(nshots=1024))
+
+
+After defining the user can perform the acquisition using
```
Comment (GabrielePalazzo) [inline doc/source/tutorials/api.rst:37]:
```suggestion
After defining, the user can perform the acquisition using
```

File: doc/source/tutorials/api.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+``experiment.acquisition`` which accepts the following parameters:
+
+* params (experiment.parameters_type): inputs parameters for the experiment
```
Comment (GabrielePalazzo) [inline doc/source/tutorials/api.rst:40]:
```suggestion
* params (experiment.parameters_type): input parameters for the experiment
```

File: doc/source/tutorials/api.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+.. code-block:: python
+
+    data, acquisition_time = experiment.acquisition(params=parameters,
+                                                    platform=platform,
+                                                    qubits=qubits)
+
+
+The user can now use the raw data acquired by the quantum processor to perform
+an arbitrary post-processing analysis. This one of the main advatanges of this API
```
Comment (GabrielePalazzo) [inline doc/source/tutorials/api.rst:56]:
```suggestion
an arbitrary post-processing analysis. This is one of the main advantages of this API
```

File: doc/source/tutorials/api.rst [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+compared to the cli execution.
+
+The fitting associate with the experiment (``experiment.fit``) can be launched in the
```
Comment (GabrielePalazzo) [inline doc/source/tutorials/api.rst:59]:
```suggestion
The fitting corresponding to the experiment (``experiment.fit``) can be launched in the
```

File: doc/source/tutorials/api.rst [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+following way:
+
+.. code-block:: python
+
+    fit, fit_time = experiment.fit(data)
+
+To be more specific the user should pass as input ``data`` which is of type
+``experiment.data_type`` and the outputs are the following:
+
+* fit: (experiment.results_type) inputs parameters for the experiment
```
Comment (GabrielePalazzo) [inline doc/source/tutorials/api.rst:69]:
```suggestion
* fit: (experiment.results_type) input parameters for the experiment
```

File: doc/source/tutorials/api.rst [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+* fit_time (float): post-processing time
+
+
+It is also possible to access the plots and the tables generated in the
+report using ``experiment.report`` which accept the following parameters:
```
Comment (GabrielePalazzo) [inline doc/source/tutorials/api.rst:74]:
```suggestion
report using ``experiment.report`` which accepts the following parameters:
```

**Review by GabrielePalazzo — COMMENTED** [review]
I just have a few comments. Feel free to modify the proposed changes.
Everything else seems good to me.

File: doc/source/tutorials/api.rst (hunk 1 above)
Comment (andrea-pasquale) [reply doc/source/tutorials/api.rst:37]: Actually here I believe that I am missing "the parameters".
```suggestion
After defining the parameters, the user can perform the acquisition using
```

**Review by GabrielePalazzo — APPROVED** [review]

File: doc/source/tutorials/api.rst [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+    parameters = experiment.parameters_type.load(dict(nshots=1024))
+
+
+After defining the parameters, the user can perform the acquisition using
+``experiment.acquisition`` which accepts the following parameters:
+
+* params (experiment.parameters_type): input parameters for the experiment
+* platform (qibolab.platform.Platform): Qibolab platform class
+* qubits (dict[QubitId, QubitPairId]) dictionary with qubits where the acquisition will run
```
Comment (scarrazza) [inline doc/source/tutorials/api.rst:42]: Could you please add a link reference for all these objects (inside the brackets)?

File: doc/source/tutorials/api.rst [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+and returns the following:
+* data (experiment.data_type): data acquired
+* acquisition_time (float): acquisition time on hardware
```
Comment (scarrazza) [inline doc/source/tutorials/api.rst:46]: The rendering is not working here, please add an empty line after the column.

**Review by scarrazza — APPROVED** [review]
Thanks, looks good, just minor fixes above.

**Pull request merged**